core/user_api.php

```diff
@@ -535,9 +535,11 @@ function user_create( $p_username, $p_password, $p_email = '',
 
 	user_ensure_name_valid( $p_username );
 	user_ensure_name_unique( $p_username );
-	user_ensure_email_unique( $p_email );
-	email_ensure_valid( $p_email );
-	email_ensure_not_disposable( $p_email );
+	if( !is_blank( $p_email ) ) {
```

Comment:
Why did you introduce this check?

After your change, blank email addresses can be set even if $g_allow_blank_email = OFF; (our default setting)

Reply:
If I remember correctly, I ran into issues upon automatic user creation, but I must have overlooked something. I'll revert this one and add a comment to SampleAuth,

Reply:
I've found the answer in SampleAuth: the reason was behind this, that Mantis will send a SignUP e-mail, if e-mail is provided (which I did not want), so I create the use with empty e-mail and assign the email afterwards. See [SampleAuth plugin](https://github.com/V1pr/SampleAuth/blob/master/core/CustomAuthPlugin.php). Anyway, I'll change the logic/doc in SampleAuth.